src/main/java/com/nfyc/studyplanservice/services/impl/TopicServiceImpl.java

```diff
@@ -53,5 +59,6 @@ public TopicDTO updateTopic(UUID topicID, TopicDTO topicDTO) {
     public void deleteTopic(UUID topicID) {
         Topic retrievedTopic = topicRepository.findById(topicID).orElseThrow(() -> new RuntimeException("Topic not found to delete"));
         retrievedTopic.getCourse().removeTopic(retrievedTopic);
+        topicRepository.save(retrievedTopic);
```

Comment:
Saving the topic is unnecessary as removing it from the course triggers orphan removal, automatically deleting the topic.

Reference: 

`@OneToMany(mappedBy = "course", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    private Set<Topic> topics = new HashSet<>();
`